pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -227,37 +236,52 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 		for _, err := range errs {
 			message = message + err.Error() + "\n"
 		}
-		workloadDegradedCondition.Status = operatorv1.ConditionTrue
-		workloadDegradedCondition.Reason = "SyncError"
-		workloadDegradedCondition.Message = message
+		workloadDegradedCondition = workloadDegradedCondition.
+			WithStatus(operatorv1.ConditionTrue).
+			WithReason("SyncError").
+			WithMessage(message)
 	} else {
-		workloadDegradedCondition.Status = operatorv1.ConditionFalse
+		workloadDegradedCondition = workloadDegradedCondition.
+			WithStatus(operatorv1.ConditionFalse)
 	}
 
 	if workload == nil {
 		message := fmt.Sprintf("deployment/%s: could not be retrieved", c.targetNamespace)
-		deploymentAvailableCondition.Status = operatorv1.ConditionFalse
-		deploymentAvailableCondition.Reason = "NoDeployment"
-		deploymentAvailableCondition.Message = message
-
-		deploymentProgressingCondition.Status = operatorv1.ConditionTrue
-		deploymentProgressingCondition.Reason = "NoDeployment"
-		deploymentProgressingCondition.Message = message
-
-		deploymentDegradedCondition.Status = operatorv1.ConditionTrue
-		deploymentDegradedCondition.Reason = "NoDeployment"
-		deploymentDegradedCondition.Message = message
+		deploymentAvailableCondition = deploymentAvailableCondition.
+			WithStatus(operatorv1.ConditionFalse).
+			WithReason("NoDeployment").
+			WithMessage(message)
+
+		deploymentProgressingCondition = deploymentProgressingCondition.
+			WithStatus(operatorv1.ConditionTrue).
+			WithReason("NoDeployment").
+			WithMessage(message)
+
+		deploymentDegradedCondition = deploymentDegradedCondition.
+			WithStatus(operatorv1.ConditionTrue).
+			WithReason("NoDeployment").
+			WithMessage(message)
+
+		// Don't overwrite workoadDegraded if it's currently true
+		if workloadDegradedCondition.Status == nil || *workloadDegradedCondition.Status == operatorv1.ConditionFalse {
```

Comment:
would it make sense to move this code to line 243 where the `workloadDegradedCondition` is set ? I think it would look like:

```
if len(errs) > 0 {
		message := ""
		for _, err := range errs {
			message = message + err.Error() + "\n"
		}
		workloadDegradedCondition = workloadDegradedCondition.
			WithStatus(operatorv1.ConditionTrue).
			WithReason("SyncError").
			WithMessage(message)
	} else if workload == nil {
		workloadDegradedCondition = workloadDegradedCondition.
			WithStatus(operatorv1.ConditionTrue).
			WithReason("NoDeployment").
			WithMessage(fmt.Sprintf("deployment/%s: could not be retrieved", c.targetNamespace))
	} else {
		workloadDegradedCondition = workloadDegradedCondition.
			WithStatus(operatorv1.ConditionFalse)
	}
```

and then we could remove setting `workloadDegradedCondition` in this branch.

Reply:
Works for me. Added to a new commit, PTAL